e2e/nomostest/client.go

```diff
@@ -99,21 +99,69 @@ func newScheme(t testing.NTB) *runtime.Scheme {
 	return s
 }
 
+// Cluster is an interface for a target k8s cluster used by the e2e tests
+type Cluster interface {
+	// Create the k8s cluster
+	Create() error
+	// Delete the k8s cluster
+	Delete() error
+	// Connect to the k8s cluster (generate kubeconfig)
+	Connect() error
+}
+
 // RestConfig sets up the config for creating a Client connection to a K8s cluster.
 // If --test-cluster=kind, it creates a Kind cluster.
 // If --test-cluster=kubeconfig, it uses the context specified in kubeconfig.
 func RestConfig(t testing.NTB, opts *ntopts.New) {
 	opts.KubeconfigPath = filepath.Join(opts.TmpDir, clusters.Kubeconfig)
 	t.Logf("kubeconfig will be created at %s", opts.KubeconfigPath)
 	t.Logf("Connect to cluster using:\nexport KUBECONFIG=%s", opts.KubeconfigPath)
+	createCluster := true
+	clusterName := opts.Name
+	opts.IsEphemeralCluster = true
+	var cluster Cluster
 	switch strings.ToLower(*e2e.TestCluster) {
 	case e2e.Kind:
-		clusters.Kind(t, *e2e.KubernetesVersion)(opts)
+		cluster = &clusters.KindCluster{
+			T:                 t,
+			Name:              clusterName,
+			KubeConfigPath:    opts.KubeconfigPath,
+			TmpDir:            opts.TmpDir,
+			KubernetesVersion: *e2e.KubernetesVersion,
+		}
 	case e2e.GKE:
-		clusters.GKECluster(t)(opts)
+		if !*e2e.CreateClusters {
```

Comment:
could this be moved out of the gke part? could this work with existing kind clusters, or other pre-existing clusters?

Reply:
Yeah we could generalize this to support providing the name of a pre-existing kind cluster.

We would need to modify the UX of the command line args a bit, as we currently use the `--gcp-cluster`/`GCP_CLUSTER` arg for passing in the name of a pre-existing GKE cluster. I figure we would probably want a single "cluster name" arg.

That being said, I'm not sure we really have a use case for providing a pre-provisioned kind cluster as it's relatively inexpensive to create a new cluster. However I do like the idea of pushing for more consistent behavior between the cluster types, as the small differences in behavior tend to become a pain point (e.g. things pass in kind and then explode when trying to run on gke)